src/components/monitorLog/ListedSpeciesTab.jsx

```diff
@@ -1,39 +1,62 @@
-import React, { useState, useCallback, useEffect } from 'react';
-import { BsPencil } from 'react-icons/bs';
+/* eslint-disable react/jsx-key */
 import {
-  Text,
-  Grid,
-  GridItem,
-  Container,
-  Tooltip,
-  HStack,
-  Table,
-  Thead,
-  Tr,
-  Th,
-  Td,
-  Tbody,
-  Box,
-  IconButton,
   Accordion,
-  AccordionItem,
   AccordionButton,
   AccordionIcon,
-  Collapse,
+  AccordionItem,
+  Box,
   Button,
-  useDisclosure,
+  Collapse,
+  Container,
+  FormControl,
+  Grid,
+  GridItem,
+  Icon,
+  IconButton,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
   Modal,
-  ModalOverlay,
+  ModalBody,
   ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  Table,
+  Tbody,
+  Td,
+  Text,
+  Tfoot,
+  Th,
+  Thead,
+  Tr,
+  useDisclosure,
+  useToast,
+  VStack,
 } from '@chakra-ui/react';
+import update from 'immutability-helper';
 import PropTypes from 'prop-types';
-import { InfoIcon } from '@chakra-ui/icons';
-import EndangeredSpeciesPopup from '../../pages/EndangeredSpeciesPopup';
+import React, { useEffect, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { FiEdit3 } from 'react-icons/fi';
+import ListedSpeciesPopup from '../ListedSpecies/ListedSpeciesPopup';
 
-const ListedSpeciesTab = ({ speciesName, speciesCode }) => {
-  const { isOpen, onOpen, onClose } = useDisclosure();
+const ListedSpeciesTab = ({ tab, speciesName, speciesCode }) => {
+  const { isOpen, onOpen: openPopup, onClose } = useDisclosure();
+  const confirmDeleteModal = useDisclosure();
   const [data, setData] = useState([]);
+  const [rowToEdit, setRowToEdit] = useState(undefined);
+  const [rowToDelete, setRowToDelete] = useState(undefined);
   const [totals, setTotals] = useState([0, 0, 0]);
+  const { register, setValue, watch } = useFormContext();
```

Comment:
⚠️ **[eslint]** <[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'register' is assigned a value but never used.